controller.rst

```diff
@@ -244,41 +237,136 @@ The Symfony templating system and Twig are explained more in the
+                      action: numberAction
+                      argument: logger
+                      # pass this specific service id
+                      id: monolog.logger.doctrine
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <!-- ... -->
+
+                <!-- Explicitly configure the service -->
+                <service id="AppBundle\Controller\LuckyController" public="true">
+                    <tag
+                        name="controller.service_arguments"
+                        action="numberAction"
+                        argument="logger"
+                        id="monolog.logger.doctrine"
+                    />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use AppBundle\Controller\LuckyController;
+        
+        $container->register(LuckyController::class)
+            ->setPublic(true)
+            ->addTag('controller.service_arguments', [
+                'action' => 'numberAction',
+                'argument' => 'logger',
+                'id' => 'monolog.logger.doctrine',
+            ])
+        ;
+
+For more information about services, see the :doc:`/service_container` article.
+
+.. note::
+    If this isn't working, make sure your controller is registered as a service,
+    :ref:`autoconfigured <services-autoconfigure>` and extends either
+    :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller` or
+    :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\AbstractController`. Or,
+    you can tag your service manually with ``controller.service_arguments``.
 
-    To get a :ref:`container configuration parameter <config-parameter-intro>`,
-    use the
-    :method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getParameter`
-    method::
+.. _accessing-other-services:
+.. _controller-access-services-directly:
 
-        $from = $this->getParameter('app.mailer.from');
+Accessing the Container Directly
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If extending the base ``Controller`` class, you can access any Symfony service
+via the :method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::get`
+method. Here are several common services you might need::
+
+    $templating = $this->get('templating');
+
+    $router = $this->get('router');
+
+    $mailer = $this->get('mailer');
+
+    // you can also fetch parameters
+    $someParameter = $this->getParameter('some_parameter');
+
+If you receive an eror like:
+
+    You have requested a non-existent service "my_service_id"
```

Comment:
This will render the error message as a block quote. Do we really want that? I suggest to use a `text` code block instead.